app/controllers/users_controller.rb

```diff
@@ -0,0 +1,21 @@
+class UsersController < ApplicationController
+  # Placeholder action for all users page
+  def index
+    @users = User.distinct
+  end
+
+  # Placeholder action for a single user page
+  def show
+    @user = User.find(params[:id])
+  end
+
+  # Placeholder action for all posts by a given user
+  def posts
+    # Your logic here
+  end
+
+  # Placeholder action for a single post page for a given user
+  def post_show
+    # Your logic here
+  end

```

Comment:
- [ ] **[OPTIONAL]** All requests related to posts should be handled by the `PostsController`, it would be great if you can delete these actions! 🤓 🔧